provider/vsphere/client.go

```diff
@@ -23,11 +23,11 @@ type Client interface {
 	Close(context.Context) error
 	ComputeResources(context.Context) ([]*mo.ComputeResource, error)
 	ResourcePools(context.Context, string) ([]*object.ResourcePool, error)
-	CreateVirtualMachine(context.Context, vsphereclient.CreateVirtualMachineParams) (*mo.VirtualMachine, error)
+	CreateVirtualMachine(context.Context, vsphereclient.CreateVirtualMachineParams, string) (*mo.VirtualMachine, error)
 	Datastores(context.Context) ([]*mo.Datastore, error)
 	DeleteDatastoreFile(context.Context, string) error
 	DestroyVMFolder(context.Context, string) error
-	EnsureVMFolder(context.Context, string) (*object.Folder, error)
+	EnsureVMFolder(context.Context, string, string) (*object.Folder, error)
```

Comment:
Actually thinking about it a bit more, I'm alright with this to stay as two paths if you prefer that, but the parent folder should come first.

Reply:
Hi @babbageclunk, indeed I need two strings: one is the vm-folder path that must not change and the other is the complementary path of folders, which I need to check one-by-one if they exist: create it otherwise. Therefore, I need two strings as inputs.